src/xpublisher.cpp

```diff
@@ -66,6 +68,46 @@ namespace xeus
                 wire_msg.send(m_controller);
                 break;
             }
+
+            if (items[2].revents & ZMQ_POLLIN)
+            {
+                // Received event: Single frame
+                // Either `1{subscription-topic}` for subscription
+                // or `0{subscription-topic}` for unsubscription
+                zmq::multipart_t wire_msg;
+                wire_msg.recv(m_publisher);
+
+                // Received event should be a single frame
+                if (wire_msg.size() != 1)
+                {
+                    throw std::runtime_error("ERROR: Received message on XPUB is not a single frame");
+                }
+
+                zmq::message_t frame = wire_msg.pop();
+                if (frame.size() == 0)
```

Comment:
Is that a valid use case? This would silently stop the publisher.

Reply:
> Is that a valid use case? 

Maybe not, I removed it.